marlowe-playground-client/src/Marlowe/Holes.purs

```diff
@@ -126,78 +129,96 @@ data Argument
   | NewtypeArg
   | GenArg MarloweType
 
-getMarloweConstructors :: MarloweType -> Map String (Array Argument)
-getMarloweConstructors ChoiceIdType = Map.singleton "ChoiceId" [ DefaultString "choiceNumber", DataArg PartyType ]
+data TermGenerator
```

Comment:
What is a TermGenerator? Could you add some comments here?

Reply:
more than generator, looking at the usage, would TermArgument be a better name?

Reply:
I was thinking this was used for generating terms from holes, that is why I called it `TermGenerator` but that may not be the whole story :/
I can add that as a comment, but for the real meaning of `Argument` type we have to ask @shmish111, I think

Reply:
so it's all about how you print a value, which can be either from a hole or from a type ahead. Either way you will get the constructor name from the editor, e.g. `"Contract"`. I understand what you mean now @palas as previously timeouts were considered as a simple or atomic argument but they weren't sum types so you would never need any type ahead or hole functionality, they were just given default values instead (either empty string or zero). The difference is between a type that has data constructors and a type that has symbolic/native constructors (strings and numbers) but now we apparently need something that is both. This gives us a problem though, imagine the hole suggestions for a timeout now, it could be either `SlotParam` or what? We could go back to `Slot` or we could offer `0` or we could not offer an option for a hard-coded timeout (in the autocomplete that is, you would still be allowed to enter one by the parser). We need to have a brief conversation about this.